src/SDK/Trace/Behavior/SpanExporterTrait.php

```diff
@@ -44,7 +56,87 @@ public function export(iterable $spans): int
             return SpanExporterInterface::STATUS_SUCCESS;
         }
 
-        return $this->doExport($spans); /** @phpstan-ignore-line */
+        // If retryPolicy is not set then just doExport once and return the status
+        if ($this->retryPolicy === null) {
+            self::logInfo('retry polict is null. Exporting the spans without retry' . PHP_EOL);
+
+            return $this->doExport($spans); /** @phpstan-ignore-line */
+        }
+
+        while ($shouldRetry) {
+            // find the delay time before retry (it is 0 for the first attempt)
+            $delay = $this->retryPolicy->getDelay($attempt);
+            if ($attempt > 0) {
+                self::logDebug('Waiting for ' . $delay . ' mili seconds before retrying export');
+                $this->retryPolicy->getDelayScheduler()->delay($delay);
+                self::logDebug('Retrying span export for ' . $attempt . ' time');
+            }
+            $attempt++;
+
+            try {
+                $status = $this->doExport($spans);
+            } catch (Throwable $e) {
+                if ($e instanceof \Error) {
+                    self::logError('Exception occured while retrying export
+                        span: ' . $e->getMessage() . '\n');
+                    $e = new RetryableExportException(
+                        $e->getMessage(),
+                        $e->getCode(),
+                        $e
+                    );
+                    $exception = $e;
+                }
+            }
+            // Find whether the request should be retried or not
+            $shouldRetry = $this->retryPolicy->shouldRetry(
+                $attempt,
+                $status,
+                $exception
+            );
+        }
+
+        return $status; /** @phpstan-ignore-line */
+    }
+
+    /**
+     * Use to define the retry policy for the exporter.
+     * Exporter needs to call it with specific retryPolicy
+     * param: RetryPolicy <RetryPolicyInterace>
+     */
+    public function setRetryPolicy(RetryPolicyInterface $retryPolicy)
+    {
+        $this->retryPolicy = $retryPolicy;
+    }
+
+    /**
+     * Returns the retry policy set by the user
+     */
+    public function getRetryPolicy()
+    {
+        return $this->retryPolicy;
+    }
+
+    /**
+     * Set the retryable status code. All the status codes apart from
+     * this list will not be retried
+     * @param array $statusCode: array of retryable status code
+     */
+    public function setRetryableStatusCodes(array $statusCode)
+    {
+        if ($this->retryPolicy) {
```

Comment:
This should check for `instanceof` not a truthy value.

Reply:
Removed method